pkg/actions/cluster/owned_test.go

```diff
@@ -0,0 +1,147 @@
+package cluster_test
+
+import (
+	"time"
+
+	"github.com/weaveworks/eksctl/pkg/kubernetes"
+	"k8s.io/client-go/kubernetes/fake"
+
+	"github.com/weaveworks/eksctl/pkg/utils/strings"
+
+	"github.com/weaveworks/eksctl/pkg/utils/tasks"
+
+	"github.com/weaveworks/eksctl/pkg/cfn/manager/fakes"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	awseks "github.com/aws/aws-sdk-go/service/eks"
+	. "github.com/onsi/ginkgo"
+	"github.com/stretchr/testify/mock"
+	"github.com/weaveworks/eksctl/pkg/actions/cluster"
+	"github.com/weaveworks/eksctl/pkg/testutils"
+
+	. "github.com/onsi/gomega"
+
+	api "github.com/weaveworks/eksctl/pkg/apis/eksctl.io/v1alpha5"
+	"github.com/weaveworks/eksctl/pkg/eks"
+	"github.com/weaveworks/eksctl/pkg/testutils/mockprovider"
+)
+
+var _ = Describe("Delete", func() {
+	var (
+		clusterName              string
+		p                        *mockprovider.MockProvider
+		cfg                      *api.ClusterConfig
+		fakeStackManager         *fakes.FakeStackManager
+		ranDeleteDeprecatedTasks bool
+		ranDeleteClusterTasks    bool
+		ctl                      *eks.ClusterProvider
+	)
+
+	BeforeEach(func() {
+		clusterName = "my-cluster"
+		p = mockprovider.NewMockProvider()
+		cfg = api.NewClusterConfig()
+		cfg.Metadata.Name = clusterName
+		fakeStackManager = new(fakes.FakeStackManager)
+		ranDeleteDeprecatedTasks = false
+		ranDeleteClusterTasks = false
+		ctl = &eks.ClusterProvider{Provider: p, Status: &eks.ProviderStatus{}}
+	})
+
+	Context("when the cluster is operable", func() {
+		It("deletes the cluster", func() {
+			//mocks are in order of being called
+			p.MockEKS().On("DescribeCluster", mock.MatchedBy(func(input *awseks.DescribeClusterInput) bool {
+				Expect(*input.Name).To(Equal(clusterName))
+				return true
```

Comment:
Generally, assertions should not be added to the matcher predicate function. The library handles presenting meaningful errors when the predicate fails. 

```suggestion
				return *input.Name == clusterName
```